cognite/client/data_classes/hosted_extractors/mappings.py

```diff
@@ -0,0 +1,292 @@
+        self.external_id = external_id
+        self.mapping = mapping
+        self.published = published
+
+    def dump(self, camel_case: bool = True) -> dict[str, Any]:
+        output = super().dump(camel_case)
+        output["mapping"] = self.mapping.dump(camel_case)
+        return output
+
+
+class MappingWrite(_MappingCore):
+    """A mapping is a custom transformation, translating the source format to a format that can be ingested into CDF.
+    Mappings are written in the Cognite transformation language.
+
+    This is the write/request format of a mapping.
+
+    Args:
+        external_id (str): The external ID provided by the client. Must be unique for the resource type.
+        mapping (CustomMapping): The custom mapping.
+        published (bool): Whether this mapping is published and should be available to be used in jobs.
+        input (InputMapping | Literal["csv", "json", "xml"]): The input mapping. Defaults to 'json'
+    """
+
+    def __init__(
+        self,
+        external_id: str,
+        mapping: CustomMapping,
+        published: bool,
+        input: InputMapping | Literal["csv", "json", "xml"] = "json",
+    ) -> None:
+        super().__init__(external_id=external_id, mapping=mapping, published=published)
+        self.input: InputMapping
+        if isinstance(input, str) and input in _INPUTMAPPING_CLASS_BY_TYPE:
+            self.input = _INPUTMAPPING_CLASS_BY_TYPE[input]()
+        elif isinstance(input, InputMapping):
+            self.input = input
+        else:
+            raise TypeError(f"Invalid input type: {input}")
+
+    @classmethod
+    def _load(cls, resource: dict[str, Any], cognite_client: CogniteClient | None = None) -> Self:
+        return cls(
+            external_id=resource["externalId"],
+            mapping=CustomMapping._load(resource["mapping"]),
+            published=resource["published"],
+            input=InputMapping._load(resource["input"]) if "input" in resource else JSONInput(),
+        )
+
+    def as_write(self) -> MappingWrite:
+        return self
+
+    def dump(self, camel_case: bool = True) -> dict[str, Any]:
+        output = super().dump(camel_case)
+        if isinstance(self.input, InputMapping):
+            output["input"] = self.input.dump(camel_case)
+        return output
+
+
+class Mapping(_MappingCore):
+    """A mapping is a custom transformation, translating the source format to a format that can be ingested into CDF.
+    Mappings are written in the Cognite transformation language.
+
+    This is the read/response format of a mapping.
+
+    Args:
+        external_id (str): The external ID provided by the client. Must be unique for the resource type.
+        mapping (CustomMapping): The custom mapping.
+        published (bool): Whether this mapping is published and should be available to be used in jobs.
+        input (InputMapping): The input mapping.
+        created_time (int): No description.
+        last_updated_time (int): No description.
```

Comment:
```suggestion
        created_time (int): The number of milliseconds since 00:00:00 Thursday, 1 January 1970, Coordinated Universal Time (UTC), minus leap seconds.
        last_updated_time (int): The number of milliseconds since 00:00:00 Thursday, 1 January 1970, Coordinated Universal Time (UTC), minus leap seconds.
```